doc/user_guide/io.rst

```diff
@@ -440,3 +442,65 @@ format stores all the captured data for each timestamp, together with a small
 header in a plain-text format. The reader extracts the measured temperature
 along the time axis, as well as the date and calibration constants stored in
 the header.
+
+
+.. _bcf-format:
+
+Bruker composite file
+----------------
+
+HyperSpy can read "hypermaps" saved with Bruker's Esprit v1.x or v2.x in bcf
+hybrid (virtual file system/container with xml and binary data, optionaly compressed) format.
+Most of relevant data extraction from bcf is implemented. Both: high
+resolution 16-bit SEM imagery and hyperspectral EDX data can be retrieved simultaniously.
+
+Note that Bruker Esprit uses the similar format for EBSD data, but even if technology
+is very similar, curently EBSD containing bcf is not supported by HyperSpy.
+
+Extra loading arguments
+^^^^^^^^^^^^^^^^^^^^^^^
+record_by: One of ('spectrum', 'image'). If specified just selected type of data
+is returned. (default None)
+
+index: index of dataset in bcf v2 files, which can hold few datasets (delaut 0)
+
+downsample: the downsample ratio of hyperspectral array (hight and width only),
+can be integer >=1, where '1' results in no downsampling (default 1). The underlying
+method of downsampling is unchangable: sum. Differently than block_reduce from skimage.measure
+it is memory efficient (does not creates intermediate arrays, works inplace).
+  
+cutoff_at_kV: if set (can be int of float >= 0) can be used either to
+crop or enlarge energy (or channels) range at max values. (default None)
+
+Example of loading reduced (downsampled, and with energy range cropped) "spectrum only"
+data from bcf (original shape: 80keV EDS range (4096 channels), 100x75 pixels):
+
+.. code-block:: python
+
+    >>> hs.load("sample80kv.bcf", record_by='spectrum', downsample=2, cutoff_at_kV=10)
+    <EDSSEMSpectrum, title: EDX, dimensions: (50, 38|595)>
+
+load the same file without extra arguments:
+
+.. code-block:: python
+
+    >>> hs.load("sample80kv.bcf")
+    [<Image, title: BSE, dimensions: (|100, 75)>,
+    <Image, title: SE, dimensions: (|100, 75)>,
+    <EDSSEMSpectrum, title: EDX, dimensions: (100, 75|1394)>]
+
+1394 energy channel range is the result of automatic shape minimalisation
+making lesser memory usage while preserving all data.
+However in conjunction with stacking, this can cause a problem.
```

Comment:
You are right: it is not clear. I don't know how to call exactly the thing. What it does? lets say output array of hyperspectra from bcf have (y, x, E) shape. Bcf contains sum EDS spectrum, so the algorithm search for last non zero value in the spectra, and sets E value in the shape to it. Now if bcf file is containing very short timed element mapping, the last non zero value in the sum spectra can be a bit different at different bcf files (stack from FIB) resulting in different shapes at E value. You made me realize that that is unnecessary stupid behavior. I implemented it because Bruker Esprit program (used to generate bcf) can be set to fixed energy ranges (10, 20, 40 or 80kv) only (that is not acceleration voltage, but range recorded in spectras regardless acceleration voltage). I think I will go back to the algorithm and change it to check acceleration voltage, compare it to the energy range (one of 10, 20, 40, 80), chose minimal and then set E to the according channel value. Then 492-494 lines will be not needed anymore.